### PR TITLE
Fix #632: <img> without "height" attribute fails to leave enough blank space for itself

### LINKED ISSUE #632: <img> without "height" attribute fails to leave enough blank space for itself
Describe the bug

**Error details**
In `write_html()`, `` without "`height`" attribute will not leave enough blank space for itself. As a result, following contents overlaps with the image.

**Minimal code**
```python
from fpdf import FPDF

pdf = FPDF()
pdf.add_page()
pdf.write_html("""<img src="/path/to/img" width="500">
<p>text</p>
""")
pdf.output("issue_632.pdf")

```

**Environment**
* OS (Linux i686)
* Python version: 3.9.2
* `fpdf2` version: Latest

If necessary, I am ready to launch a PR to fix this.

### PR DESCRIPTION
**Pull request by Bubbu0129**
Fixes #632 

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [x] A mention of the change is present in `CHANGELOG.md`

The unit test `test_img_inside_html_table_without_explicit_dimensions` in `test_html.py` isn't passed. The outcome and the original pdf looks identical, but the bytes are different somehow. Can you check it for me?

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

**Review by Lucas-C — CHANGES_REQUESTED** [review]

**Comment by Bubbu0129** [comment]
Ok, I can fix it. Here's another problem.
The memory usage plunge is due to `image()` method returning the decoded bytes of the image file. The purpose of this function is rendering the image onto the pdf, so why return the `info` dictionary? I greped the whole project and there is no file using the return value of `image()`. Removing the return value also passed all the tests.
If the return value is somehow necessary, a `flag` can be implemented in the arguments of `image()` function. If `flag == True`, then only return `rendered_width` and `rendered_height`. Otherwise, return `info`.

**Comment by Lucas-C** [comment]
> The memory usage plunge is due to `image()` method returning the decoded bytes of the image file. The purpose of this function is rendering the image onto the pdf, so why return the `info` dictionary? I greped the whole project and there is no file using the return value of `image()`. Removing the return value also passed all the tests.
> If the return value is somehow necessary, a `flag` can be implemented in the arguments of `image()` function. If `flag == True`, then only return `rendered_width` and `rendered_height`. Otherwise, return `info`.

I'm 100% OK with removing the decoded bytes from the `info` dict returned by `FPDF.image()`, with no need for an extra flag to control this behaviour (as it was never properly documented/tested), but only if this does indeed reduce the memory usage. Otherwise I don't see a need for this change.

What makes you think this behaviour of `FPDF.image()` as an impact on memory usage?
I am very curious and interested by this lead 😊

**Comment by Bubbu0129** [comment]
I first tried reserving `info` in the return value of `self.image()`, and as you can see, many tests failed due to memory overflow.
```
FAILED test/test_perfs.py::test_intense_image_rendering - Failed: Timeout >40.0s
FAILED test/image/test_load_image.py::test_share_images_cache - memunit.MemoryUsageException: Max usage 155.39Mb >= 140Mb
FAILED test/image/test_oversized.py::test_oversized_images_downscale_simple - memunit.MemoryUsageException: Max usage 155.39Mb >= 135Mb
FAILED test/image/test_oversized.py::test_oversized_images_downscale_twice - memunit.MemoryUsageException: Max usage 155.39Mb >= 135Mb
FAILED test/image/test_oversized.py::test_oversized_images_downscaled_and_highres - memunit.MemoryUsageException: Max usage 155.39Mb >= 135Mb
FAILED test/image/test_oversized.py::test_oversized_images_highres_and_downscaled - memunit.MemoryUsageException: Max usage 155.39Mb >= 135Mb
FAILED test/image/test_oversized.py::test_oversized_images_downscale_biggest_1st - memunit.MemoryUsageException: Max usage 155.39Mb >= 135Mb
FAILED test/image/test_oversized.py::test_oversized_images_downscale_biggest_2nd - memunit.MemoryUsageException: Max usage 155.39Mb >= 135Mb
```
In the original code, the return value of `self.image()` was never actually used. But in the case of issue #632, `html.py` need to obtain the `rendered_width` and `rendered_height`, so the return value is passed via the buffer. However, `info["data"]` contains the bytes of the image, plunging the memory usage.
I tried omitting `info`, and the tests got pass without memory issues. 
I haven't understood the code thoroughly. Is `info` compulsory?

**Comment by Lucas-C** [comment]
Hi and Merry Christmas @Bubbu0129 😄 

The performance tests heavily depend on your computer caracteristics:
I calibrated the threshold so that they pass in my own development environment and in GitHub Actions pipelines,
but they may fail on a computer with a slightly less performant CPU / RAM / HDD.

> In the original code, the return value of `self.image()` was never actually used. But in the case of issue #632, `html.py` need to obtain the `rendered_width` and `rendered_height`, so the return value is passed via the buffer. However, `info["data"]` contains the bytes of the image, plunging the memory usage.

OK, I think I understand your analysis, thank you for explaining.
I'm not sure about this though...
IMHO your change shouldn't have any impact, as the GC should similarly release the memory used by the image buffer once it has exited `HTML2FPDF.handle_starttag`.

FYI, I'm investigating `fpdf2` spurious memory usage in this issue: https://github.com/PyFPDF/fpdf2/issues/641

>  I haven't understood the code thoroughly. Is `info` compulsory?

What do you mean exactly?
The thing is that `FPDF.image()` used to return an `info` directory for years now,
so even if it was not properly documented, `fpdf2` users code may rely on it,
and we should, insofar as possible, try not to break their code whenever they upgrade to the latest `fpdf2` version.

Currently their is no part of `fpdf2` that rely on this `info` dict returned by `FPDF.image()` though.
If it proves to really be a performance bottleneck, I would be OK to break backward compatibility here.

**Comment by Bubbu0129** [comment]
OK, I see.
I add the `info` dict back and bumped the memory limits. 
Merry Christmas!

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,8 +24,9 @@ This can also be enabled programmatically with `warnings.simplefilter('default',
 ### Changed
 * `add_link()` creates a link to the current page by default, and now accepts optional parameters: `x`, `y`, `page` & `zoom`.
   Hence calling `set_link()` is not needed anymore after creating a link with `add_link()`.
-* `write_html()` now generates warnings for unclosed HTML tags, unless `warn_on_tags_not_matching=False` is set
+* `write_html()`now generates warnings for unclosed HTML tags, unless `warn_on_tags_not_matching=False` is set
```
Comment (Lucas-C) [inline CHANGELOG.md:27]:
```suggestion
* `write_html()` now generates warnings for unclosed HTML tags, unless `warn_on_tags_not_matching=False` is set
```

File: test/test_perfs.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 HERE = Path(__file__).resolve().parent
 
 
-@pytest.mark.timeout(40)
+@pytest.mark.timeout(60)
```
Comment (Lucas-C) [inline test/test_perfs.py:10]:
```suggestion
@pytest.mark.timeout(40)
```

**Comment by Lucas-C** [comment]
Merged! Thank you 😄

**Pull request merged**